components/browser/state/src/test/java/mozilla/components/browser/state/selector/SelectorsKtTest.kt

```diff
@@ -236,25 +236,94 @@ class SelectorsKtTest {
     }
 
     @Test
-    fun `findTabByUrl finds a regular tab with same url or returns null`() {
+    fun `findNormalOrPrivateTabByUrl finds a normal tab with same url or returns null`() {
         BrowserState(
             tabs = listOf(createTab("https://www.mozilla.org", id = "test-id"))
         ).also { state ->
-            assertNotNull(state.findTabByUrl("https://www.mozilla.org"))
+            assertNotNull(state.findNormalOrPrivateTabByUrl("https://www.mozilla.org", false))
             assertEquals(
                 "https://www.mozilla.org",
-                state.findTabByUrl("https://www.mozilla.org")!!.content.url
+                state.findNormalOrPrivateTabByUrl("https://www.mozilla.org", false)!!.content.url
             )
             assertEquals(
                 "test-id",
-                state.findTabByUrl("https://www.mozilla.org")!!.id
+                state.findNormalOrPrivateTabByUrl("https://www.mozilla.org", false)!!.id
             )
         }
 
         BrowserState(
             customTabs = listOf(createCustomTab("https://www.mozilla.org", id = "test-id"))
         ).also { state ->
-            assertNull(state.findTabByUrl("https://www.mozilla.com"))
+            assertNull(state.findNormalOrPrivateTabByUrl("https://www.mozilla.com", false))
+        }
+    }
+
+    @Test
+    fun `findNormalOrPrivateTabByUrl finds no result looking for a normal tab when only a private tab matches`() {
+        BrowserState(
+            tabs = listOf(createTab("https://www.mozilla.org", id = "test-id", private = true))
+        ).also { state ->
+            assertNotNull(state.findNormalOrPrivateTabByUrl("https://www.mozilla.org", true))
+            assertEquals(
+                "https://www.mozilla.org",
+                state.findNormalOrPrivateTabByUrl("https://www.mozilla.org", true)!!.content.url
+            )
+            assertEquals(
+                "test-id",
+                state.findNormalOrPrivateTabByUrl("https://www.mozilla.org", true)!!.id
+            )
+        }
```

Comment:
There's a problem with the test here. It says "find no result looking for a normal tab when only a private tab matches", but it asserts that it finds a normal tab.